apps/verticalhome/js/app.js

```diff
@@ -150,6 +165,206 @@
+      Array.prototype.slice.call(
+        document.querySelector('#more-apps-screen #icons').children
+      );
+
+      app.itemStore.applicationSource.
+        removeIconFromGrid( selectedMoreAppItem_data_identifier );
+
+      this.renderGrid();
+
+      if (selectedMoreAppItem.parentNode) {
+        selectedMoreAppItem.parentNode.removeChild(selectedMoreAppItem);
+      }
+
+      this.hideMoreApps();
+
+      notif.style.display = 'block';
+      notif.innerHTML = 'App added to Home Screen';
+
+      setTimeout(function() {
+        notif.style.display = 'none';
+      }, 3000);
+
+      var lastTimeSelected = app.pinnedAppsNavigation.currentIndex();
+      var insrertedIndex = app.pinnedAppsManager.items.length;
+      var scrollDiff = insrertedIndex - maxDataIndex;
+      app.pinnedAppsNavigation.elements[lastTimeSelected].classList
+        .remove('selected');
+
+      for(var z = 0; z < scrollDiff; z++){
+        var itm = app.pinnedAppsNavigation.elements[0];
+        var cln_tmp = itm.cloneNode(true);
+        app.pinnedAppsNavigation.elemList.appendChild(cln_tmp);
+        itm.parentNode.removeChild(itm);
+      }
+
+      app.pinnedAppsNavigation.elements[1].classList.add('selected');
+
+      app.pinnedAppsNavigation.scrollNum();
+
+      app.pinnedAppsNavigation.refresh();
+
+    },
+
+    pin2MoreApp: function(){
+
+      var selectedPinAppItem = document.querySelector(
+        '#pinned-apps-list .selected'
+        );
+      var selectedPinupAppItem_data_index = selectedPinAppItem
+        .getAttribute('data-index');
+
+      var notif = document.getElementById('blinknotif');
+
+      app.pinnedAppsNavigation.refresh();
+
+      this.deletedApps.push(
+        app.pinnedAppsManager.items[selectedPinupAppItem_data_index]
+        );
+
+      app.pinnedAppsManager.items[selectedPinupAppItem_data_index].
+        element.style.display = 'none';
+
+      if (selectedPinAppItem.parentNode) {
+        selectedPinAppItem.parentNode.removeChild(selectedPinAppItem);
+      }
+
+      app.pinnedAppsNavigation.elements[1].classList.add('selected');
+      app.pinnedAppsNavigation.elemList.classList.add('animation');
+      app.pinnedAppsNavigation.refresh();
+
+      notif.style.display = 'block';
+      notif.innerHTML = 'App removed from Homescreen and '+
+        'moved to "More apps"';
+
+      setTimeout(function() {
+        notif.style.display = 'none';
+      }, 3000);
+    },
+
+    personalize: function(){
+      document.getElementById('main-screen').classList
+        .add('personalize_mode');
+
+      if( !document.getElementById('addFromMoreApps')){
+        app.pinnedAppsNavigation.reset();
+
+        var plusButton = document.createElement('div');
+        plusButton.classList.add('pinned-app-item');
+        plusButton.setAttribute('id', 'addFromMoreApps');
+        plusButton.setAttribute('data-index', '998');
+        plusButton.innerHTML = '<span style="visibility: visible;" '+
+        'class="title">+</span>';
+
```

Comment:
"writing to innerHTML will cause trouble on a security audit. It is okay to set innerHTML to the empty string to delete elements, but setting this to a string, especially something that is not a string literal is considered a risk of bugs." :copyright: **(David)**

Do this through document.createElement. :smile: